data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSourceSecurityConfigurer.java

```diff
@@ -119,25 +154,130 @@ public static void setOauthProperties(final KafkaSourceConfig kafkaSourConfig,
             jass_config = jass_config.replace(";", " ");
             jass_config += String.format(extensionValue, extensionLogicalCluster, extensionIdentityPoolId);
         }
-        properties.put(SASL_JAS_CONFIG, jass_config);
+        properties.put(SASL_JAAS_CONFIG, jass_config);
     }
 
-    public static void setAwsIamAuthProperties(Properties properties, AwsIamAuthConfig awsIamAuthConfig, AwsConfig awsConfig) {
-        if (awsConfig == null) {
-            throw new RuntimeException("AWS Config is not specified");
-        }
-        properties.put("security.protocol", "SASL_SSL");
-        properties.put("sasl.mechanism", "AWS_MSK_IAM");
-        properties.put("sasl.client.callback.handler.class", "software.amazon.msk.auth.iam.IAMClientCallbackHandler");
+    public static void setAwsIamAuthProperties(Properties properties, final AwsIamAuthConfig awsIamAuthConfig, final AwsConfig awsConfig) {
+        properties.put(SECURITY_PROTOCOL, "SASL_SSL");
+        properties.put(SASL_MECHANISM, "AWS_MSK_IAM");
+        properties.put(SASL_CALLBACK_HANDLER_CLASS, "software.amazon.msk.auth.iam.IAMClientCallbackHandler");
         if (awsIamAuthConfig == AwsIamAuthConfig.ROLE) {
-            properties.put("sasl.jaas.config",
-                    "software.amazon.msk.auth.iam.IAMLoginModule required " +
-                            "awsRoleArn=\"" + awsConfig.getStsRoleArn() +
-                            "\" awsStsRegion=\"" + awsConfig.getRegion() + "\";");
+            properties.put(SASL_JAAS_CONFIG,
+                "software.amazon.msk.auth.iam.IAMLoginModule required " +
+                "awsRoleArn=\"" + awsConfig.getStsRoleArn() +
+                "\" awsStsRegion=\"" + awsConfig.getRegion() + "\";");
         } else if (awsIamAuthConfig == AwsIamAuthConfig.DEFAULT) {
-            properties.put("sasl.jaas.config",
+            properties.put(SASL_JAAS_CONFIG,
                     "software.amazon.msk.auth.iam.IAMLoginModule required;");
         }
     }
+
+    public static String getBootStrapServersForMsk(final AwsIamAuthConfig awsIamAuthConfig, final AwsConfig awsConfig) {
+        AwsCredentialsProvider credentialsProvider = DefaultCredentialsProvider.create();
+        if (awsIamAuthConfig == AwsIamAuthConfig.ROLE) {
+            String sessionName = "data-prepper-kafka-session" + UUID.randomUUID();
+            StsClient stsClient = StsClient.builder()
```

Comment:
I thought we have common way of getting credentials now. check with dlv@

Reply:
Yeah. We have to do that later.